complete/src/main/java/com/example/restservice/controller/QueueController.java

```diff
@@ -38,9 +40,18 @@
     return ResponseEntity.ok(queueService.getQueueDetails(queueId));
   }
 
-  @GetMapping(path = "/queue/status/{queueId}")
+  @GetMapping(path = "/queue/status")
   public ResponseEntity<QueueStatusResponse> getQueueStatus(
-      @PathVariable("queueId") String queueId) {
-    return ResponseEntity.ok(queueService.getQueueStatus(queueId));
+      @RequestParam(required = false) String queueId,
+      @RequestParam(required = false) String queueName) {
+    System.out.println(queueId);
+    System.out.println(queueName);
+    if (queueId != null) {
```

Comment:
This would happen even if they provided both id and name

Reply:
So, then ID takes priority. That's what I thought first. Now I am thinking, should we be validating that the queue has the same name as was requested?